hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieMetadataTableValidator.java

```diff
@@ -1136,6 +1152,11 @@ private void validateFileSlices(
           mismatch = true;
           break;
         }
+        // test for log files equality
+        if (!fileSlice1.getLogFiles().collect(Collectors.toList()).equals(fileSlice2.getLogFiles().collect(Collectors.toList()))) {
```

Comment:
do we need to sort the list before comparing?

Reply:
its already sorted. commented above.